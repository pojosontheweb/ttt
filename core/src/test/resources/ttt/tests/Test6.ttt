<%--
    JSP comment can span
    several lines
    and are ignored when template gets compiled
--%>
<%@ page import="com.xyz.myapp.Foo" %>
<%@ page extends="java.io.Serializable" %>
<%!
    Foo foo;
    int bar;
%>
<div class="foo">
    <span><%= foo.getBar() %></span>
    <ul>
    <% for(int i = 0; i < bar; i++) { %>
        <li>i equals <%= i %> </li>
    <% } %>
    </ul>
</div>