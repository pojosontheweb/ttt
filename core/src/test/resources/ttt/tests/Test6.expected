package ttt.tests;

import com.xyz.myapp.Foo;

public class Test6 extends com.pojosontheweb.ttt.Template implements java.io.Serializable {

	private final Foo foo;
	private final int bar;

	public Test6(Foo foo, int bar) {
		this.foo = foo;
		this.bar = bar;
	}

	@Override
	public void render(java.io.Writer out) throws java.io.IOException {
		write(out, "<div class=\"foo\">\n    <span>" );
		write(out,  foo.getBar()  );
		write(out, "</span>\n    <ul>\n    " );
		for(int i = 0; i < bar; i++) { 
		write(out, "\n        <li>i equals " );
		write(out,  i  );
		write(out, " </li>\n    " );
		} 
		write(out, "\n    </ul>\n</div>" );
	}
}
